azurerm/internal/services/maintenance/parse/maintenance_assignment_dedicated_host.go

```diff
@@ -0,0 +1,33 @@
+package parse
+
+import (
+	"fmt"
+	"regexp"
+
+	parseCompute "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/compute/parse"
+)
+
+type MaintenanceAssignmentDedicatedHostId struct {
+	DedicatedHostId    *parseCompute.DedicatedHostId
+	DedicatedHostIdRaw string
+	Name               string
+}
+
+func MaintenanceAssignmentDedicatedHostID(input string) (*MaintenanceAssignmentDedicatedHostId, error) {
+	groups := regexp.MustCompile(`^(.+)/providers/Microsoft\.Maintenance/configurationAssignments/([^/]+)$`).FindStringSubmatch(input)
+	if len(groups) != 3 {
+		return nil, fmt.Errorf("parsing Maintenance Assignment Dedicated Host ID: %q: Expected 3 groups", input)
```

Comment:
I think this error message wouldn't mean much to an end user that is unlikely to be familiar with how `FindStringSubmatch` works, or that it may be used here.  Perhaps something more simple?

```suggestion
		return nil, fmt.Errorf("parsing Maintenance Assignment Dedicated Host ID (%q).", input)
```